std/algebra/emulated/sw_emulated/point.go

```diff
@@ -522,60 +522,160 @@ func (c *Curve[B, S]) scalarMulGLV(Q *AffinePoint[B], s *emulated.Element[S], op
 		one := c.baseApi.One()
 		Q = c.Select(selector, &AffinePoint[B]{X: *one, Y: *one}, Q)
 	}
-	sd, err := c.scalarApi.NewHint(decomposeScalarG1, 5, s, c.eigenvalue, frModulus)
+	sd, err := c.scalarApi.NewHint(decomposeScalarG1, 7, s, c.eigenvalue, frModulus)
 	if err != nil {
 		panic(fmt.Sprintf("compute GLV decomposition: %v", err))
 	}
-	s1, s2, s3, s4 := sd[0], sd[1], sd[3], sd[4]
+	s1, s2, s3, s4, s5, s6 := sd[0], sd[1], sd[3], sd[4], sd[5], sd[6]
 
 	c.scalarApi.AssertIsEqual(
-		c.scalarApi.Add(s1, c.scalarApi.Mul(s2, c.eigenvalue)),
+		c.scalarApi.Add(s5, c.scalarApi.Mul(s6, c.eigenvalue)),
 		c.scalarApi.Add(s, c.scalarApi.Mul(frModulus, sd[2])),
 	)
-	// s1, s2 can be negative (bigints) in the hint, but are reduced mod r in
-	// the circuit. So we return in the hint both s1, s2 and s3=|s1|, s4=|s2|.
-	// In-circuit we compare s1 and s3, s2 and s4 and negate the point when a
-	// corresponding scalar is naegative.
-	selector1 := c.scalarApi.IsZero(c.scalarApi.Sub(s3, s1))
-	selector2 := c.scalarApi.IsZero(c.scalarApi.Sub(s4, s2))
-
-	var Acc, B1 *AffinePoint[B]
+
+	// s1, s2 can be negative (bigints) in the hint, but will be reduced
+	// in-circuit modulo the SNARK scalar field and not the emulated field. So
+	// we return in the hint both |s1|, |s2| and the flags s3=0/1, s4=0/1 to
+	// negate the point instead of the corresponding scalar. Since s3, s4 are
+	// either 0 or 1, we only need to check the first limb is zero or not.
+	selector1 := c.api.IsZero(s3.Limbs[0])
+	selector2 := c.api.IsZero(s4.Limbs[0])
```

Comment:
Should we add check that `(s1, s3, s5)` and `(s2, s4, s6)` correspond to each other? Otherwise we have asserted that `s5 + lambda*s6 == s`, but actually use `s1` and `s2` for binary decomposition. Malicious hint could return `s1` and `s2` which do not correspond to `s5` and `s6`.